frontend/src/ColumnSelector.js

```diff
@@ -1,80 +1,79 @@
-import React, { Component } from 'react';
-import Popover from './Popover';
-
+import React, { Component } from 'react'
+import Popover from './Popover'
 
 /**
  * Used to select the columns that appear in a table.
  */
 class ColumnSelector extends Component {
-    constructor(props) {
-        super(props);
+  constructor(props) {
+    super(props)
 
-        this.handleToggle = this.handleToggle.bind(this);
-        this.setAllColumns = this.setAllColumns.bind(this);
-        this.changeColumns = this.changeColumns.bind(this);
+    this.handleToggle = this.handleToggle.bind(this)
+    this.setAllColumns = this.setAllColumns.bind(this)
+    this.changeColumns = this.changeColumns.bind(this)
 
-        var defaultColumns = localStorage.getItem("meta-" + this.props.name);
-        if (defaultColumns) {
-            defaultColumns = JSON.parse(defaultColumns);
-        }
-        else {
-            defaultColumns = {
-                ...(props.type === "instructor" && {
-                    latest_semester: true,
-                    num_semesters: true,
-                }),
-                rInstructorQuality: true,
-                rCourseQuality: true,
-                rDifficulty: true,
-                rAmountLearned: true
-            };
-        }
-        this.defaultColumns = defaultColumns;
-        this.changeColumns(this.props.columns.map((a) => ({...a, show: a.required || !!defaultColumns[a.id]})));
+    let defaultColumns = localStorage.getItem(`meta-${this.props.name}`)
+    if (defaultColumns) {
+      defaultColumns = JSON.parse(defaultColumns)
+    } else {
+      defaultColumns = {
+        ...(props.type === 'instructor' && {
+          latest_semester: true,
+          num_semesters: true,
+        }),
+        rInstructorQuality: true,
+        rCourseQuality: true,
+        rDifficulty: true,
+        rAmountLearned: true,
+      }
     }
+    this.defaultColumns = defaultColumns
+    this.changeColumns(this.props.columns.map((a) => ({ ...a, show: a.required || !!defaultColumns[a.id] })))
+  }
 
-    changeColumns(cols) {
-        var newColumns = cols.reduce((map, obj) => {
-            map[obj.id] = obj.show;
-            return map;
-        }, {});
-        this.defaultColumns = Object.assign(this.defaultColumns, newColumns);
-        localStorage.setItem("meta-" + this.props.name, JSON.stringify(this.defaultColumns));
-        this.props.onSelect(cols);
-    }
+  changeColumns(cols) {
+    const newColumns = cols.reduce((map, obj) => {
+      map[obj.id] = obj.show
+      return map
+    }, {})
```

Comment:
ooooh wow this is a smart way of doing dis

Reply:
Eric is big brain